python/ray/tests/test_basic.py

```diff
@@ -864,6 +864,69 @@ def method(self):
     assert ray.get([id1, id2, id3, id4]) == [0, 1, "test", 2]
 
 
+def test_many_fractional_resources(shutdown_only):
+    ray.init(num_cpus=2, num_gpus=2, resources={"Custom": 2})
+
+    @ray.remote
+    def g():
+        return 1
+
+    @ray.remote
+    def f(block, accepted_resources):
+        true_resources = {
+            resource: value[0][1]
+            for resource, value in ray.get_resource_ids().items()
+        }
+        if block:
+            ray.get(g.remote())
+        return true_resources == accepted_resources
+
+    # Check that the resource are assigned correctly.
+    for rand1, rand2, rand3 in np.random.uniform(size=(100, 3)):
```

Comment:
Isn't this going to be really slow since we're doing `ray.get` inside of the loop? One option is just to put the assert inside of the remote function so you don't need to check the return value.